elastic/kibana
# [Exploratory View] add a global filter to exploratory view when there is only one series

## Summary

Fixes https://github.com/elastic/kibana/issues/119933

This PR improves the user experience of breakdowns within a single series. Breakdowns use a `top_values` metric. Previously, all the most relevant `top_values` result would be displayed, regardless of whether the value should have been filtered out based on the user input. This led to a bloated legend (or y axis, in the case of core web vitals which shows breakdowns along the y axis) with lots of irrelevant breakdown values that the user did not care to see.

This PR solves that issue by applying a global filter to lens embeddable. Global filters are processed before the `top_values` metric, ensuring that unnecessary `top_values` results are not shown. 

Global filters are only able to be applied when there is a single series. Otherwise, the global filter could conflict filters in one of the series. For this reason, global filters are only applied when there is a single series, and removed when the user adds a second series. When this happens, you'll notice irrelevant legend items return. This is a current limitation in Lens, which does not yet allow for layer-level (or in our case series level) filters. 

Core Web Vitals Report Type
--
Before
![image](https://user-images.githubusercontent.com/11356435/148136340-bd9a1de4-53e9-4935-a1bc-aec36529dbee.png)

After
![image](https://user-images.githubusercontent.com/11356435/146967794-3d3dfdb4-ec0f-4c0e-95ec-8dafd3244899.png)

KPI Over Time or Data Distribution Report Types
--
Before (Extra irrelevant items in the sidebar, regardless of single or multi series).
![image](https://user-images.githubusercontent.com/11356435/148136383-acc9992e-de74-44d5-b894-067f49a8832f.png)

After (Irrelevant items removed from the sidebar in single series. Behavior remains the same in multi series).
![image](https://user-images.githubusercontent.com/11356435/148773355-bd5b3a5b-a897-40cc-9f29-7d10da04db78.png)

### Testing
Core Web Vitals Report Type
--
1. Navigate to exploratory view with some UX data available. The easiest way to do this is to connect your local Kibana instance to edge via the `oblt-cli` tool.
2. Select the `Core Web Vitals` report type, and configure the report type and metric type of your choice.
3. Apply a breakdown to the data. For example, you could choose to breakdown by web application.
4. Apply a filter relevant to the breakdown metric. For example, if we broke down by web application we'll want to them choose to filter for a specific web application, or filter a specific web application out.
5. Click Apply
6. Observe that the filtered out data does not appear in the y axis

Data distribution or KPI over time Report Type
--
1. Navigate to exploratory view with some UX data available. The easiest way to do this is to connect your local Kibana instance to edge via the `oblt-cli` tool.
2. Select the Data distribution or KPI Over time report type, and configure the report type and metric type of your choice.
3. Apply a breakdown to the data. For example, you could choose to breakdown by web application.
4. Apply a filter relevant to the breakdown metric. For example, if we broke down by web application we'll want to them choose to filter for a specific web application or filter a specific web application out.
5. Click Apply
6. Observe that the filtered out data does not appear in the left-hand side legend
